mne/epochs.py

```diff
@@ -1957,22 +1958,59 @@ def apply_function(
         if dtype is not None and dtype != self._data.dtype:
             self._data = self._data.astype(dtype)
 
+        args = getfullargspec(fun)[0] + getfullargspec(fun)[4]
+        if channel_wise is False:
+            if ("ch_idx" in args) or ("ch_name" in args):
+                raise ValueError(
+                    "apply_function cannot access ch_idx or ch_name "
+                    "when channel_wise=False"
+                )
+        elif ("ch_idx" in args) and ("ch_name" in args):
+            raise ValueError(
+                "apply_function cannot access both ch_idx and ch_name. pick one!"
+            )
```

Comment:
yes, @larsoner asked the same..
@drammock and @sappelhoff had argued for prohibiting this option to avoid potential trouble. 
further up I also added a check that raises an error if both are requested, so elif -> if wouldnt make any difference here.

I am happy to change the code if you settle this amongst you :)
it's probably not too relevant anyway - difficult to imagine use cases where one would need both ch index AND name..

Reply:
after a live discussion with @dominikwelke I've come around to the idea that it's really not dangerous to allow the user function to have access to both `ch_idx` and `ch_name`. And it simplifies our implementation slightly to not constrain them to only use one.